Загрузил файл со скриптом для создания базы данных.

thisfozz opened the pull request:
Для того чтобы создать базу данных необходимо для начала выполнить следующую команду. Это нужно делать перед созданием самих таблиц.

Эта команда установит расширение и откроет доступ к типу данных uuid и функции для его генерации `uuid_generate_v4()`

`CREATE EXTENSION IF NOT EXISTS "uuid-ossp";`

thisfozz committed:
Загрузил файл со скриптом для создания базы данных.

thisfozz commented:
![](https://github.trello.services/images/mini-trello-icon.png) [QF. Design Database Schema (№ 13)](https://trello.com/c/qpziHFOs/19-qf-design-database-schema-%E2%84%96-13)

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+CREATE TABLE SecretQuestions 
```
название файла все-таки сделай соответствующим. Я имею ввиду расширение смени с txt на sql

thisfozz replied:
Исправлено

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+    CONSTRAINT user_id_fk FOREIGN KEY(user_id) REFERENCES Users(user_id) ON DELETE CASCADE
+);
+
+CREATE TABLE Packages
```
Удаление пакетов не планируется? Может добавить soft delete?

thisfozz replied:
Удаление пакетов самим пользователем не планируется. За такой фичей он может обратиться к администратору.

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+    round_id_pk SERIAL PRIMARY KEY,
+    package_id INTEGER NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    type_rounds VARCHAR(7) NOT NULL CHECK(type_rounds ILIKE 'Обычный' or type_rounds ILIKE 'Финал'),
```
я бы назвал round_type

thisfozz replied:
Исправлено

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+
+CREATE TABLE Rounds
+(
+    round_id_pk SERIAL PRIMARY KEY,
```
в принципе нет смысла писать для PK таблицы user_id, round_id. В названии таблицы уже есть упоминание, что это. Я бы оставил просто id

thisfozz replied:
Исправлено

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+    secret_answer_hash VARCHAR(256) NOT NULL, 
+    TwoFA_auth_method TEXT,
+	Google_Authorization BYTEA,
+    CONSTRAINT secret_question_fk FOREIGN KEY(selected_secret_question_id) REFERENCES SecretQuestions(secret_q_id) ON DELETE CASCADE
```
при удалении типа секрета каскадом грохнешь всех юзеров? думаешь, так лучше будет?

thisfozz replied:
Исправлено

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+    question_text TEXT,
+    question_theme VARCHAR(15),
+    question_image_url BYTEA,
+    question_cost INTEGER NOT NULL CHECK(question_cost >= 1),
```
может сделать ограничение по верхней границе тоже?

thisfozz replied:
Лишняя головная боль для составителя вопросов. Максимум простора для творчества для самого создателя пакета.

lexeg commented on Quizzify/Quizzify/Scripts/Quzzify.txt:
```diff
@@ -0,0 +1,61 @@
+    question_comment TEXT,
+    answer_text TEXT,
+    answer_image_url BYTEA,
+    CONSTRAINT questions_round_id_fk FOREIGN KEY (round_id) REFERENCES Rounds(round_id_pk) ON DELETE CASCADE,
```
проверь аккуратно еще везде по таблицам: где действительно нужны каскады, а где NULL, например сделать.

lexeg reviewed (approved):
Апрув поставил авансом. Посмотри комментарии.

thisfozz commented:
![](https://github.trello.services/images/mini-trello-icon.png) [QF. Transfer Database Script to GitHub](https://trello.com/c/IKmNIQlB/25-qf-transfer-database-script-to-github)

thisfozz committed:
Был изменен формат скрипта с .txt на .sql
Проведена работа над наименованием строк.
Было убрано каскадное удаление вопроса при удалении User. Теперь при удалении User таблица SecretQuestions останется без изменений и наоборот при удалении значени из SecretQuestions - поле selected_secret_question_id у User станет NULL.

CozyWool commented:
Посмотрел, вроде все сделано, что мы хотели на прошлом занятии
